### ISSUE TITLE
[a11y]: Issues with the buttons used by the ProgressIndicator component

### ISSUE THREAD

**Issue by rachel-orti** [carbon-design-system/carbon#20762]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.89.0 and latest version (v1.93.0)

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

By using the storybook, we have the following 2 potential violations on the Progress Indicator component with the IBM Equal Access Accessibility Checker (see the attached screen shots).
These issues seem to be real violations because:

- The steps are represented by buttons that are not clickable and not tabbable.
- The buttons contain a paragraph (see attached screenshot) which is supposed to be ignored by browsers.

<img width="1664" height="1290" alt="Image" src="https://github.com/user-attachments/assets/4c1b7c4e-eda2-48f3-a321-8da648470914" />

<img width="1896" height="380" alt="Image" src="https://github.com/user-attachments/assets/6be5e3de-fb28-452e-9fa1-00f2da8945de" />

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?globals=&args=&id=components-progressindicator--default&viewMode=story

### Steps to reproduce

You can use the storybook here: https://react.carbondesignsystem.com/iframe.html?globals=&args=&id=components-progressindicator--default&viewMode=story
 Ensure to have the IBM Equal Access Accessibility Checker plugin installed in your browser, then run the Accessibility Assessment by clicking "Scan".

<img width="1632" height="852" alt="Image" src="https://github.com/user-attachments/assets/01315292-cffa-4925-8b24-2beffbe6ec45" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by rafael-nogueras** [comment]
@heloiselui Can we get an ETA on this? [One of our A11Y issues](https://github.ibm.com/WatsonOrchestrate/wo-tracker/issues/44717) is blocked because of it. If a fix became available ASAP we could still try to incorporate it. Thanks!

**Comment by heloiselui** [comment]
Hello @rafael-nogueras, I don’t have an exact date yet, but I’ll take a look at this during this sprint and keep this issue updated as progress is made.

**Comment by rafael-nogueras** [comment]
@heloiselui Thanks! :-)
